lib/collection/src/shard.rs

```diff
@@ -46,33 +55,193 @@ pub struct Collection {
     schema_store: Arc<SchemaStorage>,
 }
 
-/// Collection holds information about segments and WAL.
-impl Collection {
+/// Shard holds information about segments and WAL.
+impl Shard {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
-        segments: Arc<RwLock<SegmentHolder>>,
+        segment_holder: SegmentHolder,
         config: CollectionConfig,
-        wal: Arc<Mutex<SerdeWal<CollectionUpdateOperations>>>,
-        mut update_handler: UpdateHandler,
-        runtime_handle: Runtime,
-        path: PathBuf,
+        wal: SerdeWal<CollectionUpdateOperations>,
+        optimizers: Arc<Vec<Arc<Optimizer>>>,
+        collection_path: &Path,
         schema_store: Arc<SchemaStorage>,
     ) -> Self {
+        let segment_holder = Arc::new(RwLock::new(segment_holder));
+
+        let blocking_threads = if config.optimizer_config.max_optimization_threads == 0 {
+            max(num_cpus::get() - 1, 1)
+        } else {
+            config.optimizer_config.max_optimization_threads
+        };
+        let optimize_runtime = runtime::Builder::new_multi_thread()
+            .worker_threads(2)
+            .thread_name_fn(|| {
+                static ATOMIC_ID: AtomicUsize = AtomicUsize::new(0);
+                let id = ATOMIC_ID.fetch_add(1, Ordering::SeqCst);
+                format!("optimizer-{}", id)
```

Comment:
Ideally we should name the the optimizer threads so that it is clear where they belong.

Maybe something like 
```rust
format!("optimizer-{}-shard-{}-thread-{}", collection_name, shard_id, id)
```

Do you plan on giving shards a unique ID in a later PR?
What do you think about the proposed thread naming?